monitoring/src/listener/listener.rs

```diff
@@ -0,0 +1,54 @@
+use networking::p2p::network_channel::{NetworkChannelMsg, NetworkChannelRef, NetworkChannelTopic};
+use riker::actor::*;
+use std::fs::File;
+use crate::listener::records::Record;
+use std::io::Write;
+use log::*;
+
+type NetworkListenerRef = ActorRef<NetworkChannelMsg>;
+
+pub struct NetworkListener {
+    storage: File,
+    network_channel: NetworkChannelRef,
+}
+
+impl NetworkListener {
```

Comment:
Name suggests that thsi actor is listening to a network (connection?). Maybe use some other name which will be more descriptive of what the actor does.

Reply:
Actor does indeed listen to the Network messages sent on the NetworkChannel. NetworkChannelListener was chosen as more descriptive name.